File: src/main/java/com/lessmarkup/engine/data/TableMetadata.java
package com.lessmarkup.engine.data;

import com.lessmarkup.framework.helpers.PropertyDescriptor;
import com.lessmarkup.framework.helpers.TypeHelper;
import java.util.HashMap;
import org.atteo.evo.inflector.English;

class TableMetadata {
    
    private final String name;
    private final HashMap<String, PropertyDescriptor> columns;
    
    public TableMetadata(Class<?> sourceType) {
        this.name = English.plural(sourceType.getSimpleName());
        this.columns = new HashMap<>();
        
        for (PropertyDescriptor property : TypeHelper.getProperties(sourceType)) {
            this.columns.put(property.getName(), property);
        }
    }
    
    public String getName() {
        return this.name;
    }
    
    public HashMap<String, PropertyDescriptor> getColumns() {
        return this.columns;
    }
}

File: src/main/scala/com/lessmarkup/engine/data/TableMetadata.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.engine.data

import com.lessmarkup.framework.helpers.TypeHelper
import org.atteo.evo.inflector.English

class TableMetadata(sourceType: Class[_]) {

  private val name: String = English.plural(sourceType.getSimpleName)
  private val columns = TypeHelper.getProperties(sourceType).map(p => (p.getName, p)).toMap

  def getName: String = {
    this.name
  }

  def getColumns = this.columns
}
